js/ui/calendar/ui.calendar.selection.strategy.js

```diff
@@ -0,0 +1,227 @@
+        this.updateAriaSelected(value, previousValue);
+
+        !this._currentDateChanged && this.calendar._initCurrentDate();
+        this._currentDateChanged = false;
+    },
+
+    getDefaultCurrentDate: noop,
+
+    _getLowestDateInArray: function(dates) {
+        if(dates.length) {
+            return new Date(Math.min(...dates));
+        }
+    },
+
+    _convertToDate: function(value) {
+        return this.calendar._convertToDate(value);
+    },
+
+    _isMaxZoomLevel: function() {
+        return this.calendar._isMaxZoomLevel();
+    },
+
+    _updateViewsOption: function(optionName, optionValue) {
+        this.calendar._updateViewsOption(optionName, optionValue);
+    },
+
+    _updateViewsValue: function(value) {
+        this._updateViewsOption('value', value);
+    },
+
+    _updateCurrentDate: function(value) {
+        this.calendar.option('currentDate', value || new Date());
+    }
+});
+
+const CalendarSingleSelectionStrategy = CalendarSelectionStrategy.inherit({
+    NAME: 'SingleSelection',
+
+    getViewOptions: function() {
+        return {
+            value: this.dateOption('value'),
+            range: [],
+            selectionMode: 'single',
+        };
+    },
+
+    selectValue: function(selectedValue, e) {
+        this.dateValue(selectedValue, e);
+    },
+
+    updateAriaSelected: function(value, previousValue) {
+        value = value ?? [this.dateOption('value')];
+        previousValue = previousValue ?? [];
+
+        this.callBase(value, previousValue);
+    },
+
+    getDefaultCurrentDate: function() {
+        return this.dateOption('value');
+    },
+
+    _updateViewsValue: function(value) {
+        this._updateViewsOption('value', value[0]);
+    },
+});
+
+const CalendarMultiSelectionStrategy = CalendarSelectionStrategy.inherit({
+    NAME: 'MultiSelection',
+
+    getViewOptions: function() {
+        return {
+            value: this.dateOption('values'),
+            range: [],
+            selectionMode: 'multi',
+        };
+    },
+
+    selectValue: function(selectedValue, e) {
+        const values = [...this.dateOption('values')];
+        const alreadySelectedIndex = values.findIndex(date => date?.toDateString() === selectedValue.toDateString());
+
+        if(alreadySelectedIndex > -1) {
+            values.splice(alreadySelectedIndex, 1);
+        } else {
+            values.push(selectedValue);
+        }
+
+        this._updateCurrentDate(selectedValue);
+        this._currentDateChanged = true;
+        this.dateValue(values, e);
+    },
+
+    updateAriaSelected: function(value, previousValue) {
+        value = value ?? this.dateOption('values');
+        previousValue = previousValue ?? [];
+
+        this.callBase(value, previousValue);
+    },
+
+    getDefaultCurrentDate: function() {
+        return this._getLowestDateInArray(this.dateOption('values').filter(value => value));
+    },
+});
+
+const CalendarRangeSelectionStrategy = CalendarSelectionStrategy.inherit({
```

Comment:
shouldn't we inherit it from MultiSelectionStrategy to reduce copypaste?

Reply:
I don't think it's will reduce copypaste, range and multi methods are pretty different